opentech/apply/users/templates/users/login.html

```diff
@@ -1,15 +1,65 @@
+
 {% block header_modifier %}header--light-bg{% endblock %}
 {% block page_title %}Login{% endblock %}
 {% block title %}Login{% endblock %}
 
 {% block content %}
   <div class="wrapper wrapper--small">
+    {% if wizard.steps.current == 'auth' %}
+        <p>{% blocktrans %}Enter your credentials.{% endblocktrans %}</p>
+    {% elif wizard.steps.current == 'token' %}
+        {% if device.method == 'call' %}
+            <p>{% blocktrans %}We are calling your phone right now, please enter the
+                digits you hear.{% endblocktrans %}</p>
+        {% elif device.method == 'sms' %}
+            <p>{% blocktrans %}We sent you a text message, please enter the tokens we
+                sent.{% endblocktrans %}</p>
+        {% else %}
+            <p>{% blocktrans %}Please enter the tokens generated by your token
+                generator.{% endblocktrans %}</p>
+        {% endif %}
+    {% elif wizard.steps.current == 'backup' %}
+        <p>{% blocktrans %}Use this form for entering backup tokens for logging in.
+            These tokens have been generated for you to print and keep safe. Please
+            enter one of these backup tokens to login to your account.{% endblocktrans %}</p>
+    {% endif %}
+
     <form class="form form--with-p-tags" method="post">
       {% csrf_token %}
-      {{ form.as_p }}
-      <p><a class="link link--small" href="{% url 'users:password_reset' %}">Forgot your password?</a></p>
-      <button class="link link--button-secondary" type="submit">Login</button>
+      {{ wizard.management_form }}
+
+      {% if wizard.steps.current == 'auth' %}
+        {{ form.as_p }}
+        <p><a class="link link--small" href="{% url 'users:password_reset' %}">Forgot your password?</a></p>
+        <button class="link link--button-secondary" type="submit">Login</button>
+      {% else %}
+        {{ wizard.form }}
+
+        {# hidden submit button to enable [enter] key #}
+        <div style="margin-left: -9999px"><input type="submit" value=""/></div>
+
+        {% if other_devices %}
+            <p>{% trans "Or, alternatively, use one of your backup phones:" %}</p>
+            <p>
+                {% for other in other_devices %}
+                <button name="challenge_device" value="{{ other.persistent_id }}"
+                                class="btn btn-default btn-block" type="submit">
+                    {{ other|device_action }}
+                </button>
+            {% endfor %}</p>
+        {% endif %}
+        {% if backup_tokens %}
+            <p>{% trans "As a last resort, you can use a backup token:" %}</p>
+            <p>
+                <button name="wizard_goto_step" type="submit" value="backup"
+                                class="btn btn-default btn-block">{% trans "Use Backup Token" %}</button>
+            </p>
+        {% endif %}
+
+        {% include "two_factor/_wizard_actions.html" %}
+      {% endif %}
     </form>
 
     <div class="wrapper wrapper--inner-space-large">
```

Comment:
Login in OTF email appears in Token page as well while login. We may want to include it only in initial form as a choice between username and password. Also, need to test how it will work with new 2fa feature.

Reply:
Good catch, added a check to fix this now.